flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/agg/DistinctAggCodeGen.scala

```diff
@@ -955,4 +956,9 @@ class DistinctAggCodeGen(
        """.stripMargin
     }
   }
+
+  override def setWindowSize(generator: ExprCodeGenerator): String = {
+    throw new TableException(
+      "Distinct shouldn't set window size, this is a bug, please file a issue.")
```

Comment:
Is it possible that `COUNT(DISTINCT) OVER ...` may hit this?

Reply:
No, currently, the dictinct + over is not supported, see the [code](https://github.com/apache/flink/blob/bf342d2f67a46e5266c3595734574db270f1b48c/flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/batch/BatchPhysicalOverAggregateRule.scala#L209).
Also, in this pr, it'll [skip][https://github.com/apache/flink/pull/19727/files#diff-04af992db6ef031313241dbb2fc6d1bc6bf8b75b8b7b98d2effeb101fbe17e36R870) `setWindowSize` when it's  for `DistinctAggCodeGen`.